lib/temporal/workflow/context.rb

```diff
@@ -215,14 +215,53 @@ def wait_for_all(*futures)
         return
       end
 
-      def wait_for(future)
+      # Block workflow progress until any future is finished or any unblock_condition
+      # block evaluates to true.
+      def wait_for(*futures, &unblock_condition)
+        if futures.empty? && unblock_condition.nil?
+          raise 'You must pass either a future or an unblock condition block to wait_for'
+        end
+
         fiber = Fiber.current
+        should_yield = false
+        blocked = true
+
+        if futures.any?
+          if futures.all?(&:finished?)
```

Comment:
This should probably be `futures.any?(&:finished?)`, right? At least judging by the example and a test implementation it waits for a single future before proceeding

Reply:
Found another similar issue here where if the condition already evaluates to true, we should not block on the futures, yield, or register any dispatch handlers for them